lib/Core/Site/Site.php

```diff
@@ -25,32 +72,110 @@ class Site implements SiteInterface
     private $loadService;
 
     /**
-     * @param \Netgen\EzPlatformSiteApi\API\FilterService $filterService
-     * @param \Netgen\EzPlatformSiteApi\API\FindService $findService
-     * @param \Netgen\EzPlatformSiteApi\API\LoadService $loadService
+     * @param \eZ\Publish\API\Repository\ContentService $contentService
+     * @param \eZ\Publish\API\Repository\ContentTypeService $contentTypeService
+     * @param \eZ\Publish\API\Repository\FieldTypeService $fieldTypeService
+     * @param \eZ\Publish\API\Repository\LocationService $locationService
+     * @param \eZ\Publish\API\Repository\SearchService $searchService
+     * @param \eZ\Publish\API\Repository\SearchService $synchronousSearchService
+     * @param string[] $prioritizedLanguages
+     * @param bool $useAlwaysAvailable
      */
     public function __construct(
-        FilterServiceInterface $filterService,
-        FindServiceInterface $findService,
-        LoadServiceInterface $loadService
+        ContentService $contentService,
+        ContentTypeService $contentTypeService,
+        FieldTypeService $fieldTypeService,
+        LocationService $locationService,
+        SearchService $searchService,
+        SearchService $synchronousSearchService,
+        $useAlwaysAvailable
     ) {
-        $this->filterService = $filterService;
-        $this->findService = $findService;
-        $this->loadService = $loadService;
+        $this->contentService = $contentService;
+        $this->contentTypeService = $contentTypeService;
+        $this->fieldTypeService = $fieldTypeService;
+        $this->locationService = $locationService;
+        $this->searchService = $searchService;
+        $this->synchronousSearchService = $synchronousSearchService;
+        $this->useAlwaysAvailable = $useAlwaysAvailable;
+    }
+
+    /**
+     * Setter for prioritized languages configuration, used to update
+     * the configuration on scope change.
+     *
+     * @param array $prioritizedLanguages
+     */
+    public function setPrioritizedLanguages(array $prioritizedLanguages)
+    {
+        $this->prioritizedLanguages = $prioritizedLanguages;
     }
 
     public function getFilterService()
     {
+        if ($this->filterService !== null) {
+            return $this->filterService;
+        }
+
+        $this->filterService = new FilterService(
+            $this->getDomainObjectMapper(),
+            $this->synchronousSearchService,
+            $this->contentService,
+            $this->prioritizedLanguages,
```

Comment:
Well, here's the solution, though I'm not sure what to do with it, maybe it should be preventively burned, just in case: https://github.com/netgen/ezplatform-site-api/pull/35/commits/bd36bb85efd2e473fe935152c6f7c1397bde5074